common/src/main/java/com/revenuecat/purchases/common/offlineentitlements/ProductEntitlementMapping.kt

```diff
@@ -1,49 +1,50 @@
 package com.revenuecat.purchases.common.offlineentitlements
 
+import com.revenuecat.purchases.utils.optNullableString
 import org.json.JSONArray
 import org.json.JSONObject
 
 data class ProductEntitlementMapping(
-    val mappings: List<Mapping>
+    val mappings: Map<String, Mapping>
 ) {
     companion object {
-        private const val PRODUCTS_KEY = "products"
-        private const val PRODUCT_ID_KEY = "id"
+        private const val PRODUCT_ENTITLEMENT_MAPPING_KEY = "product_entitlement_mapping"
+        private const val PRODUCT_ID_KEY = "product_identifier"
+        private const val BASE_PLAN_ID_KEY = "base_plan_id"
         private const val ENTITLEMENTS_KEY = "entitlements"
 
         fun fromJson(json: JSONObject): ProductEntitlementMapping {
-            val productsArray = json.getJSONArray(PRODUCTS_KEY)
-            val mappings = mutableListOf<Mapping>()
-            for (productIndex in 0 until productsArray.length()) {
-                val productObject = productsArray.getJSONObject(productIndex)
+            val productsObject = json.getJSONObject(PRODUCT_ENTITLEMENT_MAPPING_KEY)
+            val mappings = mutableMapOf<String, Mapping>()
+            for (mappingIdentifier in productsObject.keys()) {
+                val productObject = productsObject.getJSONObject(mappingIdentifier)
                 val productIdentifier = productObject.getString(PRODUCT_ID_KEY)
+                val basePlanId = productObject.optNullableString(BASE_PLAN_ID_KEY)
                 val entitlementsArray = productObject.getJSONArray(ENTITLEMENTS_KEY)
                 val entitlements = mutableListOf<String>()
                 for (entitlementIndex in 0 until entitlementsArray.length()) {
                     entitlements.add(entitlementsArray.getString(entitlementIndex))
                 }
-                mappings.add(Mapping(productIdentifier, entitlements))
+                mappings[mappingIdentifier] = Mapping(productIdentifier, basePlanId, entitlements)
             }
             return ProductEntitlementMapping(mappings)
         }
     }
 
     data class Mapping(
-        val identifier: String,
+        val productIdentifier: String,
+        val basePlanId: String?,
         val entitlements: List<String>
     )
 
-    fun toMap(): Map<String, List<String>> {
-        return mappings.associateBy({ it.identifier }, { it.entitlements })
-    }
```

Comment:
Can't really use this now that it's not a simple mapping anymore, so I removed it